pandas/core/arrays/categorical.py

```diff
@@ -675,20 +691,12 @@ def from_codes(cls, codes, categories, ordered=False):
             if msg:
                 raise ValueError(msg)
 
-        try:
-            codes = coerce_indexer_dtype(codes, categories)
-        except (ValueError, TypeError):
-            raise ValueError(
-                "codes need to be convertible to an arrays of integers")
-
```

Comment:
``fastpath`` is  always True in this constructor, and ``coerce_indexer_dtype`` is already called in the __init__ constructor when ``fastpath=True``, so calling here is not needed.